src/bundle.rs

```diff
@@ -190,6 +212,12 @@ pub trait Authorization: fmt::Debug {
     type SpendAuth: fmt::Debug + Clone;
 }
 
+/// Defines the authorization type of an Orchard bundle with a proof.
+pub trait AuthorizedWithProof: Authorization<SpendAuth = redpallas::Signature<SpendAuth>> {
```

Comment:
why do we need a new trait?

Why not add `proof()` to `impl ActionGroupAuthorized {` ?

Reply:
In serailization normal ZSA bundle and ZSA bundle that is inside ActionGroup are almost identical, except Authorization. It's Authorized and ActionGroupAuthorized. To depuplicate serialization I need them both to implement a trait that has a proof() method

Reply:
not sure I follow. 
`AuthorizedWithProof` adds no new information to the content. There is already a proof inside. Why we need new generic `AuthorizedWithProof` again?

Reply:
Both ActionGroup and OrchardZSA budnles are Bundle<A, V, D>. They are identical, but in frist case A=Authorized, B=ActionGroupAuthorized. During serialization we need to write a proof which is a part of both, but it is not a part of any common trait, it is just 2 unrelated fields in 2 unrelated structures. So to deduplicate the code I added this common interface that describes something that is Authorized and has proof method

Reply:
<img width="765" alt="image" src="https://github.com/user-attachments/assets/e71b002f-bad5-40b5-a3f9-c0a657299cab" />

In both cases `A` is `Authorization`
let's add `fn proof(&self) -> Option<&Proof>;` to  `Authorization` and implement it 4 times (`InProgress` and `Unauthorized` will return `None`).

Better than more adding traits with partial intersection.